plugins/catalog-backend/api-report.md

```diff
@@ -180,7 +180,7 @@ export type CatalogEnvironment = {
   database: PluginDatabaseManager;
   config: Config;
   reader: UrlReader;
-  permissions: PermissionAuthorizer;
+  permissions: PermissionEvaluator;
```

Comment:
I'm a bit torn on this, as this is really a breaking change. What do you think about expressing this as a deprecation? Thinking `PermissionAuthorizer | PermissionEvaluator` along with the two separate implementations, with the old one being deprecated

Reply:
Shouldn't this case already been covered?
If someone bumps backstage version,  catalog-backend is going to be bumped, including the changes in this PR.
But, at the same time also `plugin-permission-node` is going to be bumped, which means `ServerPermissionClient` is going to implement `PermissionEvaluator`.
Of course, if someone provides a custom implementation of `PermissionAuthorizer`, it is going to break.
I wonder if marking the packages as breaking could be enough, given that most likely adopters wouldn't have provided any custom implementation.
What do you think?

Reply:
Indeed, custom implementations would be broken. The catalog backend is at 1.0 and this is a stable API. [Our promise](https://backstage.io/docs/overview/versioning-policy#package-versioning-policy) to the community for that is this:

> Breaking changes to stable exports include a deprecation phase if possible. The deprecation must have been released for at least one mainline release before it can be removed.

I think it's possible to provide a deprecation path in this case, so I think we should